marklogic-data-hub-central/ui/src/util/authorities.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+
+export interface IAuthoritiesContextInterface {
+    authorities: string[];
+    setAuthorities: (authorities: string[]) => void;
+    canReadMapping: () => boolean;
+    canWriteMapping: () => boolean;
+    canReadMatchMerge: () => boolean;
+    canWriteMatchMerge: () => boolean;
+    canReadLoadData: () => boolean;
+    canWriteLoadData: () => boolean;
+    canReadEntityModel: () => boolean;
+    canWriteEntityModel: () => boolean;
+    canReadFlow: () => boolean;
+    canWriteFlow: () => boolean;
+    canReadStepDefinition: () => boolean;
+    canWriteStepDefinition: () => boolean;
+    hasOperatorRole: () => boolean;
+}
+
+/**
+ *  Authorities Service that has authorities read in at login
+ *
+ * @description This service provides information about what capabilities the user has.
+ */
+export class AuthoritiesService implements IAuthoritiesContextInterface {
+    public authorities: string[] = [];
+
+    public setAuthorities: (authorities: string[]) => void = (authorities: string[]) => {
+        this.authorities = authorities;
+    };
+
+    public canReadMapping:() => boolean = () => {
```

Comment:
Could the UI just use "readMapping" as well? It seems simpler and less effort to not have to map e.g. "readMapping" to "canReadMapping".

Reply:
@rjrudin I was wondering if we need 2 authorities, 1 each for read and write. Wouldnt canWrite*=false would mean its a read only authority. Please clarify.

Reply:
I think having one for read and one for write avoids confusion. Either you have the ability to do something - you have the authority - or you don't.

Reply:
Thanks!

Reply:
I'm not too strongly opposed to "readMapping". I just thought I'd leave "canReadMapping" as it makes clear it is a method that returns a boolean, rather than performing an action like "reading a mapping", whatever that action would entail.

@wooldridge @xnikhil08  Any preference here?

Reply:
Yeah, I'll withdraw my comment here - I think it's fine to do this in the UI, particularly if that makes the code read more nicely.